src/hud_element.c

```diff
@@ -40,94 +50,653 @@ Vec3s gHudElementSizes[26] = {
+            if (texSizeY > 16) {
+                lry = baseY + drawSizeY - 1;
+            } else {
+                lry = baseY + drawSizeY;
+            }
+            isLastTileY = TRUE;
+        }
+
+        isLastTileX = FALSE;
+        uls = 0;
+        ulx = baseX;
+        while (TRUE) {
+            lrx = ulx + 1024.0 / widthScale * 64.0;
+            lrs = uls + 63;
+            if (flipX) {
+                texStartX = texSizeX;
+            } else {
+                texStartX = 0;
+            }
+
+            if (lrx < 0  || ulx > SCREEN_WIDTH) {
+                break;
+            }
+
+            if (lrx >= SCREEN_WIDTH) {
+                s32 temp = uls + SCREEN_WIDTH + 63;
+                temp -= baseX + lrs;
+                lrs = temp - 1;
+
+                lrx = SCREEN_WIDTH;
+                isLastTileX = TRUE;
+            }
+
+            if (lrs + 1 >= texSizeX) {
+                lrs = texSizeX - 1;
+                if (texSizeX > 16) {
+                    lrx = baseX + drawSizeX - 1;
+                } else {
+                    lrx = baseX + drawSizeX;
+                }
+                isLastTileX = TRUE;
+            }
+
+            gDPPipeSync(gMasterGfxPos++);
+
+            if (isLastTileX ) {
+                tileMode = !isLastTileY;
+            }
+            if (!isLastTileX && !isLastTileY) {
+                tileMode = 0;
+            }
+            if (!isLastTileX && isLastTileY) {
+                tileMode = 2;
+            }
+            if (isLastTileX && isLastTileY) {
+                tileMode = 3;
+            }
+
+            switch (fmt) {
+                case 0:
+                    gDPSetCombineMode(gMasterGfxPos++, G_CC_DECALRGBA, G_CC_DECALRGBA);
+                    if ((hudElement->flags & HUD_ELEMENT_FLAGS_TRANSPARENT) == 0) {
+                        gDPSetCombineMode(gMasterGfxPos++, G_CC_DECALRGBA, G_CC_DECALRGBA);
+                    } else {
+                        gDPSetCombineLERP(gMasterGfxPos++, 0, 0, 0, TEXEL0, PRIMITIVE, 0, TEXEL0, 0, 0, 0, 0, TEXEL0, TEXEL0, 0, PRIMITIVE, 0);
+                    }
+
+                    if (hudElement->flags & HUD_ELEMENT_FLAGS_TRANSPARENT) {
+                        gDPSetPrimColor(gMasterGfxPos++, 0, 0, 0, 0, 0, hudElement->opacity);
+                    }
+
+                    if (!flipX && !flipY) {
+                        gDPLoadTextureTile(gMasterGfxPos++, imageAddr, G_IM_FMT_RGBA, G_IM_SIZ_32b, texSizeX, texSizeY, uls, ult, lrs, lrt, 0,
+                                            G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, masks, maskt, G_TX_NOLOD, G_TX_NOLOD);
+                    } else {
+                        gDPLoadTextureTile(gMasterGfxPos++, imageAddr, G_IM_FMT_RGBA, G_IM_SIZ_32b, texSizeX, texSizeY, uls, ult, lrs, lrt, 0,
+                                            G_TX_MIRROR | G_TX_WRAP, G_TX_MIRROR | G_TX_WRAP, masks, maskt, G_TX_NOLOD, G_TX_NOLOD);
+                    }
+                    break;
+
```

Comment:
We don't have empty lines between switch cases elsewhere in the codebase. I think it's okay for larger cases, but we should be consistent probably

Reply:
(relevant before this point too, just saw it a few times and decided to comment here)